pkg/session/session.go

```diff
@@ -2417,6 +2417,12 @@ func runStmt(ctx context.Context, se *session, s sqlexec.Statement) (rs sqlexec.
 	if err != nil {
 		return nil, err
 	}
+	if sessVars.TxnCtx.CouldRetry && !s.IsReadOnly(sessVars) {
+		// Only when the txn is could retry and the statement is not read only, need to do stmt-count-limit check.
```

Comment:
Maybe a link to https://docs.pingcap.com/tidb/stable/tidb-configuration-file#stmt-count-limit is more clear to explain `CouldRetry`.

Reply:
done.